dataflow_test.py

```diff
@@ -452,3 +452,33 @@ def test_slice_raises_ValueError(args):
 @mark.xfail
 def test_pipes_must_end_in_a_sink():
     raise NotImplementedError
+
+
+def test_count_filter_filters_correctly():
+    even_numbers = list(range(0, 20, 2))
+    odd_numbers  = list(range(1, 20, 2))
+    the_source   = odd_numbers + even_numbers
+
+    odd      = df.count_filter(lambda n: n % 2)
+    filtered = []; the_sink = df.sink(filtered.append)
+
+    df.push(source = the_source,
+            pipe   = df.pipe(odd.filter, the_sink))
+
+    assert filtered == odd_numbers
+
+
+def test_count_filter_counts_correctly():
```

Comment:
I'm tempted to unite `test_count_filter_{counts,filters}_correctly`.

+ The two aspects are very closely tied in this component
+ There's lots of repetition between the two tests, making for a suboptimal signal to noise ratio.